item-spec/examples/sentinel2-sample.json

```diff
@@ -55,6 +55,15 @@
       ]
     ]
   },
+  "properties": {
+    "collection": "s2-l2a",
+    "datetime": "2018-06-05T10:00:27.464Z",
+    "platform": "sentinel-2",
+    "gsd": 10,
+    "proj:epsg": 32635,
+    "eo:cloud_cover": 88.459539,
```

Comment:
Some fields are required, but not listed (e.g. eo:gsd)

Reply:
? It looks like eo:gsd moved to just be 'gsd' in the [instrument section] of common metadata. https://github.com/radiantearth/stac-spec/blob/dev/item-spec/common-metadata.md#instrument 

This does highlight though that it's no longer required - could be good for EO extension to make it required. That should probably be another issue.

Reply:
It seems I shouldn't make reviews at 1AM. You are right, eo:gsd has been moved to core, but then still eo:bands is required (at least at the moment of writing this, will likely change soon).

Reply:
So what's my action on this one then?

Reply:
Add a `eo:bands` object to properties and make sure the `eo:bands` references in the assets are valid.

Reply:
Ok, I looked into if I could do this easily, but I think it's beyond the scope of this PR for me. It looks like landsat also doesn't define its eo:bands references. And it seems like it'll change soon. I can add a ticket that says 'update item spec examples to have valid bands' or something.

Reply:
In this case I'd simply remove all eo fields altogether, but at least the eo:bands in the assets, because that's really broken and confusing without having the referenced band array.

Reply:
See #797 for a "fix".